packages/editor/src/components/CodeEditor/StateViewer.tsx

```diff
@@ -36,17 +36,17 @@ const style = css`
 
 export const StateViewer: React.FC<{ store: Record<string, unknown> }> = ({ store }) => {
   const [filterText, setFilterText] = useState('');
-  const [refreshFlag, setRefreshFlag] = useState(false);
+  const [refreshFlag, setRefreshFlag] = useState(0);
   const data = useMemo(() => {
     return pickBy(store, (_v, key) => {
       return filterText ? key.includes(filterText) : true;
     });
-  // eslint-disable-next-line react-hooks/exhaustive-deps
+    // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [store, filterText, refreshFlag]);
 
   useEffect(() => {
     const stop = watch(store, () => {
-      setRefreshFlag(v => !v);
+      setRefreshFlag(v => v + 1);
```

Comment:
Why can this change fix the bug?

Reply:
We use a vue proxy to listen to the store, and if we `mergeState` again during an update, it will trigger two updates
I think react's `SyntheticEvent` will merge the two updates (setRefreshFlag twice and return the initial value) so that the view is not updated    
Here is an example(data is updated, but the view is not): https://stackblitz.com/edit/react-dmnplm?file=src/App.js